src/v3.ts

```diff
@@ -0,0 +1,39 @@
+import prettier from 'prettier';
+import { OpenAPI3, PropertyMapper } from './types';
+
+export const WARNING_MESSAGE = `/**
+ * This file was auto-generated by swagger-to-ts.
+ * Do not make direct changes to the file.
+ */
+`;
+
+export const PRETTIER_OPTIONS: prettier.Options = { parser: 'typescript', singleQuote: true };
+
+export default function generateTypesV3(schema: OpenAPI3, propertyMapper?: PropertyMapper): string {
+  if (!schema.components || !schema.components.schemas) {
+    throw new Error(`🧐 Components Object missing from schema https://swagger.io/specification`);
+  }
+
+  // 1st pass: expand $refs first to reduce lookups & prevent circular refs
+  const expandedRefs = JSON.parse(JSON.stringify(schema.components.schemas), (_, node) => {
+    // TODO: if a node has a $ref, then convert "#/components/schemas/<$ref>" -> components['schemas']['<$ref>']
+
+    return node;
+  });
+
+  // 2nd pass: support primitives
+  const primitives = JSON.parse(JSON.stringify(expandedRefs), (_, node) => {
+    // TODO: convert `{ some_key: { type: "string" } }` -> `{ some_key: string }`
```

Comment:
2nd todo: convert the following objects to the following strings:

| Original                                        | Transformed |
|:--------------------------------|:--------------|
| `{ some_key: { type: 'string' } }` | `{ some_key: string }` |
| `{ some_key: { type: 'number' } }` |  `{ some_key: number }` |
| `{ some_key: { type: 'integer' } }` |  `{ some_key: number }` (same as `number`) |
| `{ some_key: { type: 'boolean' } }` |  `{ some_key: boolean }` |